frontend/src/components/map/LeafletMap.tsx

```
@@ -327,23 +333,283 @@ const SearchControl: React.FC<{
+// Create a component for the map nodes
+const MapNodes: React.FC<{ onLoadingChange: (state: LoadingState) => void }> = ({ onLoadingChange }) => {
+  const map = useMap();
+  const [nodes, setNodes] = useState<MapNode[]>([]);
+  const markersRef = useRef<L.Marker[]>([]);
+
+  // Validate node data
+  const isValidNode = (node: MapNode): boolean => {
+    return !!(
+      node &&
+      node.siteDetails?.approximate_latitude &&
+      node.siteDetails?.approximate_longitude &&
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)